cmake/Docs.cmake

```diff
@@ -0,0 +1,27 @@
+find_package(Doxygen REQUIRED)
```

Comment:
oh wow! your whole pipeline is so clean! 
I am scared of docker containers, and never had to set up them myself. But your code sells them very well.

Reply:
Thanks 👍 I think there is still room for improvement but by using docker we can indeed remove a lot of the configuration and env setup from the CI itself.